cmd/poller/poller.go

```diff
@@ -575,6 +576,56 @@ func (me *Poller) loadCollector(class, object string) error {
 	return nil
 }
 
+// Union2 merges the fields of a Poller with the fields of a node.
+// This is a way to bridge the struct world with the string typed world.
+// If one of the poller field's does not exist in hNode, it will be copied
+// from poller to hNode.
+// If the field already exists in hNode, nothing is copied.
+// Instead of comparing each field of the poller individually and being forced
+// to keep this method in sync with the Poller struct, reflection via yaml marshaling
+// is used to do the comparison. First the poller is marshaled to yaml and then
+// unmarshalled into a list of generic yaml node. Each generic yaml node is walked, checking
+// if there is a corresponding node in hNode, when there isn't one, a new hNode is created
+// and populated with the yaml node's content. Finally, the new hNode is added to its parent
+
+func Union2(hNode *node.Node, poller *conf.Poller) {
```

Comment:
Thanks comments are very helpful. I agree reflection brings better maintenance.
Are we only considering implementing struct till poller level or zapi also?
If we want to replace all then this union method isn't needed.

Reply:
Struct for all config related things initially.  Maybe zapis later.
ZAPI is a much bigger challenge - best would be to take zapidoc and generate structs, similar to KFS.
There are some big downsides with that approach though, one of the biggest being you end up with a struct per ONTAP version.